crates/python/qcs_sdk/compiler/quilc.pyi

```diff
@@ -148,10 +148,10 @@ class RandomizedBenchmarkingRequest:
     def qubits(self, value: int): ...
     @property
     def gateset(self) -> List[str]:
-        """List of Quil programs, each describing a Clifford."""
+        """Sequence of Quil programs, each describing a Clifford."""
```

Comment:
These are still technically returning a `List` though, right?

What's the motivation behind moving from `List` to `Sequence` in general though - are some callers using tuples/sets?

Reply:
Oh yeah, that should be `List`, I think a heavy handed find and replace changed the docstring here, switched it back.

My general approach is to accept interfaces/abstract types like `Sequence` or `Mapping` as input arguments and return concrete types like `List` or `Dict`.

As input types, abstract types like `Sequence` and `Mapping`s are more flexible since they define the minimum required logic needed. As you mention, a `Sequence` allows for list, tuples, and sets and per the pyo3 docs, it will convert `Sequence`s to `Vec`s and `Mapping` to `HashMap`s, so we aren't misrepresenting the input types we can accept. The `Sequence` and `Mapping` types are also immutable, so we're expressing that we won't modify the input arguments via the type as well. In practice, I have found this flexibility pretty helpful, since there are places throughout pyQuil where we use list and tuples interchangeably.

The idea behind returning concrete types has an inverse justification, they have at least parity with their corresponding abstract type, but often define a superset of functionality, so when I get one of those types back, I know exactly what I'm getting and what I can do with it.